app/table-streams/synonyms/index.ts

```diff
@@ -34,26 +34,59 @@ async function removeIndex(id: string) {
   }
 }
 
-async function putIndex(synonym: Synonym) {
+async function putIndex(synonymGroup: SynonymGroup) {
   const client = await getSearchClient()
   await client.index({
     index,
-    body: synonym,
+    id: synonymGroup.synonymId,
+    body: synonymGroup,
   })
 }
 
+async function getIndex(synonym: Synonym) {
+  const client = await getSearchClient()
+  try {
+    const group = await client.get({
+      index: 'synonym-groups',
+      id: synonym.synonymId,
+    })
+
+    return group.body._source as SynonymGroup
+  } catch (e) {
+    return null
+  }
+}
+
 export const handler = createTriggerHandler(
   async ({ eventName, dynamodb }: DynamoDBRecord) => {
-    const id = unmarshallTrigger(dynamodb!.Keys).id as string
+    if (!eventName || !dynamodb) return
     const promises = []
-
-    if (eventName === 'REMOVE') {
+    const id = unmarshallTrigger(dynamodb!.Keys).synonymId
+    const synonym = unmarshallTrigger(dynamodb!.NewImage) as Synonym
+    const existingGroup = await getIndex(synonym)
```

Comment:
Remember that we are treating DynamoDB as the source of truth, not OpenSearch. Please get the list of synonyms matching the UUID of the altered record using DynamoDB.